File: c/src/kidsWithCandies_test.c
#include <assert.h>
#include <stdlib.h>
#include <string.h>

#include "kidsWithCandies.c"

typedef struct test {
    int* candies;
    int candiesSize;
    int extraCandies;
    bool* want;
} test;

int main(int argc, char* argv[])
{
    test tests[] = {
        { (int[]) { 2, 3, 5, 1, 3 }, 5, 3, (bool[]) { true, true, true, false, true } },
        { (int[]) { 4, 2, 1, 1, 2 }, 5, 1, (bool[]) { true, false, false, false, false } },
        { (int[]) { 12, 1, 12 }, 3, 10, (bool[]) { true, false, true } },
    };
    size_t len_tests = sizeof(tests) / sizeof(test);

    for (size_t i = 0; i < len_tests; i++) {
        test* t = &tests[i];
        int returnSize = 0;
        bool* got = kidsWithCandies(t->candies, t->candiesSize, t->extraCandies, &returnSize);
        assert(returnSize == t->candiesSize);
        assert(memcmp(got, t->want, sizeof(bool) * returnSize) == 0);
        free(got);
    }

    exit(EXIT_SUCCESS);
}

File: python/src/kidsWithCandies_test.py
from typing import List

import pytest

from kidswithCandies import Solution

solution = Solution()


@pytest.mark.parametrize(
    "candies, extraCandies, want",
    (
        ([2, 3, 5, 1, 3], 3, [True, True, True, False, True]),
        ([4, 2, 1, 1, 2], 1, [True, False, False, False, False]),
        ([12, 1, 12], 10, [True, False, True]),
    ),
)
def test_kidsWithCandies(candies: List[int], extraCandies: int, want: List[bool]):
    got = solution.kidsWithCandies(candies, extraCandies)
    assert got == want
